### ISSUE TITLE
Can we use Element.classList or not?

### ISSUE THREAD

**Issue by RunDevelopment** [PrismJS/prism#1815]
I just want to ask whether we use [`Element.classList`](https://developer.mozilla.org/en-US/docs/Web/API/Element/classList) or not. Cross-browser support is not perfect but it should be good enough.

I'm asking because I thought that we don't use `.classList` but plugins like Line numbers are sending [mixed](https://github.com/PrismJS/prism/blob/master/plugins/line-numbers/prism-line-numbers.js#L88) [messages](https://github.com/PrismJS/prism/blob/master/plugins/line-numbers/prism-line-numbers.js#L147), so I'm not sure.

/cc @mAAdhaTTah

**Comment by ExE-Boss** [comment]
Well, as&nbsp;long as&nbsp;we&nbsp;don’t use it&nbsp;on&nbsp;SVG elements and avoid using the&nbsp;`replace()` method, we&nbsp;should be&nbsp;fine.

**Comment by Golmote** [comment]
We've been trying to stay compatible with IE9 for quite some time I believe (although not _actively_ trying AFAICT), at least for prism-core.
I think we could benefit from upgrading our code to support more recent browsers, but this would require bumping Prism to the next major version.
I haven't had the chance to look at browser usage worldwide for a while, I believe Prism is used by quite a bunch of Chinese projects, so we should be careful about this.

**Comment by Golmote** [comment]
Oh, and now I just saw an issue where we speak about dropping IE11 support. Haha we might not be compatible with IE9 at all, actually. I don't know anymore. A major version with explicit browser support (at least in the readme) would be great.

**Comment by ExE-Boss** [comment]
There’s [this polyfill for `classList`](https://www.npmjs.com/package/classlist-polyfill).

**Comment by Golmote** [comment]
True, but I'm not a fan of adding polyfills to core features. If we are to drop old IE compatibility then we don't have to bother too much about this (but we could list this polyfill in an appropriate section of the readme). And if we were to keep old IE compatibility, adding this polyfill would still require a major version bump.
